src/d3d9/d3d9_device.cpp

```diff
@@ -4314,21 +4315,54 @@ namespace dxvk {
     auto convertFormat = pResource->GetFormatMapping().ConversionFormatInfo;
 
     if (likely(convertFormat.FormatType == D3D9ConversionFormat_None)) {
-      VkExtent3D texLevelExtentBlockCount = util::computeBlockCount(levelExtent, formatInfo->blockSize);
-      D3D9BufferSlice slice = AllocTempBuffer<false>(copyBuffer->info().size);
+      const DxvkFormatInfo* formatInfo = imageFormatInfo(pResource->GetFormatMapping().FormatColor);
+      VkImageSubresourceLayers dstLayers = { VK_IMAGE_ASPECT_COLOR_BIT, subresource.mipLevel, subresource.arrayLayer, 1 };
+
+      const D3DBOX& box = pResource->GetDirtyBox(subresource.arrayLayer);
+      VkOffset3D scaledBoxOffset = {
+        int32_t(alignDown(box.Left  >> subresource.mipLevel, formatInfo->blockSize.width)),
+        int32_t(alignDown(box.Top   >> subresource.mipLevel, formatInfo->blockSize.height)),
+        int32_t(alignDown(box.Front >> subresource.mipLevel, formatInfo->blockSize.depth))
+      };
+      VkExtent3D scaledBoxExtent = util::computeMipLevelExtent({
+        uint32_t(box.Right - box.Left),
```

Comment:
Is this already aligned to block size when stored?

Reply:
No. We handle that in line 4332. Not entirely sure whether that's good enough considering we align down the offset.
Should probably use that to calculate the extent.